### PR TITLE
October work

### PR DESCRIPTION
**Pull request by andyneff**
- Fix instance name for singularity instances

### CODE REVIEW

File: linux/compat.bsh [hunk 1]
```diff
@@ -150,7 +150,39 @@ function load_vsi_compat()
...
+  #**
+
+  # Make life easier (and quicker after 6 uses)
+  local bash_major_version="${BASH_VERSINFO[0]}"
+  local bash_minor_version="${bash_major_version}${BASH_VERSINFO[1]}"
+
+  #**
+  # .. var:: bash_feature_posix_process_substitution
+  #
+  # In bash 5.0 and older, process substitution isn't allows when in posix mode. This issue can arise if a function contains non-posix commands and is exported, before calling ``bash --posix``. Bash will complain about each and every function that is non-posix.
```
Comment (scott-vsi) [inline linux/compat.bsh:162]: allowed

Comment (scott-vsi) [reply linux/compat.bsh:162]: exported before

File: linux/compat.bsh [hunk 2]
```diff
@@ -150,7 +150,39 @@ function load_vsi_compat()
...
+  #
+  # :Value: * ``0`` - Bash does support parameter transformations (Bash 4.4 or newer)
```
Comment (scott-vsi) [inline linux/compat.bsh:164]: Here you say 4.4 of newer, but in the summary, you said 5.0 or older

File: linux/compat.bsh [hunk 3]
```diff
@@ -261,40 +289,58 @@ function load_vsi_compat()
...
+  #
+  # .. code-block:: bash
+  #
+  #    exec {BASH_XTRACEFD}>"somefile"
+  #
+  # ``BASH_XTRACEFD``'s value is set, but whatever triggers ``BASH_XTRACEFD`` to work, is not triggered.
+  #
+  # .. note::
+  #
+  #   ``BASH_XTRACEFD`` and automatically allocation file descriptors are not available features in bash 4.0 or older, but :var:`bash_bug_allocate_xtracefd` will still be set to ``0``
```
Comment (scott-vsi) [inline linux/compat.bsh:305]: automatically allocated file (?)

File: linux/compat.bsh [hunk 4]
```diff
@@ -261,40 +289,58 @@ function load_vsi_compat()
...
   # .. var:: bash_feature_declare_global
   #
   # Does bash support declaring global variables with the ``declare`` command.
   #
   # :Value: * ``0`` - Bash supports ``declare -g``
   #         * ``1`` - Bash does not support ``declare -g``
+  #
+  # .. var:: bash_behavior_all_traps_display_inherited
+  #
+  # When a subshell is spawned, the ERR/RETURN/DEBUG traps always show up in ``trap -p`` even if there are not actually inherited. (``set -E`` and ``set -T`` control whether these traps are actually inherited or not). This is an annoying behavior bash has and it makes determining which traps are actually active, impossible to determine. Starting in bash 4.2, all signals started observing this behavior. This was done on purpose, but really makes tracking traps harder.
```
Comment (scott-vsi) [inline linux/compat.bsh:335]: even if they are not

Comment (scott-vsi) [reply linux/compat.bsh:335]: actually active impossible to determine

File: linux/compat.bsh [hunk 5]
```diff
@@ -391,7 +446,7 @@ function load_vsi_compat()
...
   #
-  # With most bash 3.2's, using process substitution for the BASH_ENV env variable does not work as expected. (Perhaps the file is read twice, and since it is a stream, it cannot seek to the beginning.) Work around is to write to a temporary file instead. Some bash 3.2 do support this. Since this is expensive to test (one fork, which is only expensive in windows), it is not evaluated into a variable on load. Instead, this is a function that is called. The result is cached so that is does not need to be tested again. The feature has less to do with the exact version of bash, and how it is compiled or patched.
+  # With most bash 3.2's, using process substitution for the BASH_ENV env variable does not work as expected. (For some reason.) Work around is to write to a temporary file instead. Some bash 3.2 do support this. Since this is expensive to test (one fork, which is only expensive in windows), it is not evaluated into a variable on load. Instead, this is a function that is called. The result is cached so that is does not need to be tested again. The feature has less to do with the exact version of bash, and how it is compiled or patched.
```
Comment (scott-vsi) [inline linux/compat.bsh:449]: work as expected (for some reason).

Comment (scott-vsi) [reply linux/compat.bsh:449]: version of bash, but rather how it is compiled

Comment (scott-vsi) [reply linux/compat.bsh:449]: Some bash 3.2's do support this.

File: linux/compat.bsh [hunk 6]
```diff
@@ -409,6 +464,42 @@ function load_vsi_compat()
     return "${__bash_bug_bash_env_process_substitution}"
   }
 
+  #**
+  # .. function:: bash_feature_rcfile
+  #
+  # Depending on how bash is compiled (``-DSYS_BASHRC``), bash will use or ignore the ``--rcfile`` argument. This has to be tests by actually seeing if this works, so it is a function that is called on demand, instead of on :file:`compat.bsh` load.
```
Comment (scott-vsi) [inline linux/compat.bsh:470]: This has to be tested by actually

File: linux/just_files/singularity_functions.bsh [hunk 7]
```diff
@@ -122,7 +122,7 @@ function singularity_env_pass()
 #           * ``singular_environment`` - A key-value pair array of environment variables that need to be passed to the singularity container.
 #           * ``singular_image`` - The filename of the singularity image used.
 #           * ``singular_mount_points`` - List of mount points that need to be created when importing a docker image to singularity. This is only important for older systems that do not support overlayfs (like CentOS 6). Directories must end in ``/``, and files must not.
-#           * ``singular_instance_name`` - The name the singularity instance uses, including a unique hash for the project (Based off of username and path)
+#           * ``singular_instance_name`` - The name of the singularity instance uses, including a unique hash for the project (Based off of username and path)
```
Comment (scott-vsi) [inline linux/just_files/singularity_functions.bsh:125]: no `of`

File: linux/requirements.bsh [hunk 8]
```diff
@@ -305,7 +268,7 @@ function version_gt()
 #
 # Splits a version string up into its elements and stores them in an array.
 #
-# The last element is always the pre-release/metadata. If there is no pre-release or metadata, the last element will be an empty string.
+# Any non-numerical values are stored in the last element is as pre-release/metadata (including leading/trailing whitespace)
```
Comment (scott-vsi) [inline linux/requirements.bsh:271]: last element as pre-release

File: linux/requirements.bsh [hunk 9]
```diff
@@ -317,11 +280,39 @@ function version_gt()
...
+  else
+    dynamic_set_a "${1}" ${BASH_REMATCH[1]} "${BASH_REMATCH[2]}"
+  fi
+}
+
+#**
+# .. function:: split_version_string_and_remainder
+#
+# :Arguments: - ``$1`` - Variable name to store the array in
+#             - ``$2`` - Variable name to leftover versioning information in (anything non-numeric)
```
Comment (scott-vsi) [inline linux/requirements.bsh:301]: name to store leftover

File: linux/requirements.bsh (hunk 8 above)
Comment (scott-vsi) [reply linux/requirements.bsh:271]: Is this true here, or only in `split_version_string_and_remainder`

Comment (scott-vsi) [reply linux/requirements.bsh:271]: nm. i gather that it is true here as well

File: linux/compat.bsh (hunk 2 above)
Comment (andyneff) [reply linux/compat.bsh:164]: It also says "parameter transformation", bad copy and paste I suspect

File: linux/compat.bsh (hunk 3 above)
Comment (andyneff) [reply linux/compat.bsh:305]: It's another bash feature, dictated by `bash_feature_allocate_file_descriptor`

```
 Each  redirection  that  may  be  preceded by a file descriptor number may instead be preceded by a word of the form {varname}.  In this case, for each redirection operator except >&- and <&-, the shell will allocate a file descriptor greater than or equal to 10 and assign it to varname.  If >&- or <&- is preceded by {varname}, the value of varname defines the file descriptor to close.   If {varname} is supplied, the redirection persists beyond the scope of the command, allowing the shell programmer to manage the file descriptor himself.
```

File: linux/requirements.bsh (hunk 8 above)
Comment (andyneff) [reply linux/requirements.bsh:271]: Yeah, the two functions are fairly similar, but it was useful to have two different behaviors to cover all the different scenarios properly, without adding extra complications.

**Pull request merged**